src/pretix/control/views/event.py

```diff
@@ -400,6 +405,78 @@ def post(self, request, *args, **kwargs):
             return self.get(request)
 
 
+class MailSettingsPreview(EventPermissionRequiredMixin, View):
+    permission = 'can_change_settings'
+
+    # return the origin text if key is missing in dict
+    class SafeDict(dict):
+        def __missing__(self, key):
+            return '{' + key + '}'
+
+    @cached_property
+    def items(self):
+        return {
+            'mail_text_order_placed': ['total', 'currency', 'date', 'invoice_company',
+                                       'event', 'paymentinfo', 'url', 'invoice_name'],
+            'mail_text_order_paid': ['event', 'url', 'invoice_name', 'invoice_company', 'payment_info'],
+            'mail_text_order_free': ['event', 'url', 'invoice_name', 'invoice_company'],
+            'mail_text_resend_link': ['event', 'url', 'invoice_name', 'invoice_company'],
+            'mail_text_resend_all_links': ['event', 'orders'],
+            'mail_text_order_changed': ['event', 'url', 'invoice_name', 'invoice_company'],
+            'mail_text_order_expire_warning': ['event', 'url', 'expire_date', 'invoice_name', 'invoice_company'],
+            'mail_text_waiting_list': ['event', 'url', 'product', 'hours', 'code']
+        }
+
+    @cached_property
+    def dummy_data(self):
+        return {
+            'event': self.request.event.name,
+            'total': 100,
+            'currency': self.request.event.currency,
+            'date': date_format(now() + timedelta(days=7), 'SHORT_DATE_FORMAT'),
```

Comment:
got it! I realised that the problem comes from `@cached_property`. Since I cached the placeholder `{date}` and `{expire_date}` as `str`, the date value didn't change accordingly.